src/dfx/src/lib/telemetry.rs

```diff
@@ -1,22 +1,35 @@
+#![allow(unused)] // remove when there are no more todos
+
+use crate::config::dfx_version;
 use crate::lib::error::DfxResult;
 use crate::CliOpts;
 use anyhow::Context;
 use clap::parser::ValueSource;
 use clap::{ArgMatches, Command, CommandFactory};
 use dfx_core::config::directories::project_dirs;
+use dfx_core::fs;
+use fd_lock::RwLock as FdRwLock;
+use semver::Version;
+use serde::Serialize;
 use std::ffi::OsString;
+use std::fs::OpenOptions;
+use std::io::Write;
 use std::path::PathBuf;
 use std::sync::{Mutex, OnceLock};
+use std::time::Instant;
 
 static TELEMETRY: OnceLock<Mutex<Telemetry>> = OnceLock::new();
+// separate var to preserve Telemetry as Eq
+static START_TIME: OnceLock<Instant> = OnceLock::new();
```

Comment:
This can go as an `elapsed_ms: Option<u128>` in the Telemetry struct.  Main can calculate the elapsed time and set it by calling a `Telemetry::set_elapsed_time(elapsed)` method just like any of the other fields that we're reporting. There's no need for a OnceLock, because it will all happen on the stack in main(). Tests won't set the elapsed time, so they won't have to concern themselves with differences related to it.